src/radio-group.ts

```diff
@@ -195,51 +192,89 @@ export default class GlideCoreRadioGroup
   }
 
   override firstUpdated() {
-    if (this.disabled) {
-      for (const radio of this.#radioElements) {
-        radio.disabled = true;
+    for (const radio of this.#radioElements) {
+      radio.privateRequired = this.required;
+      radio.tabIndex = -1;
+
+      if (this.disabled) {
+        radio.disabled = this.disabled;
       }
     }
 
-    if (this.required) {
-      for (const radio of this.#radioElements) {
-        radio.privateRequired = true;
+    // There's nothing to stop developers from adding a `checked` attribute to more
+    // than one radio. How native handles it with `<select>` is to choose the last
+    // selected option. This mimics that behavior, which seems reasonable.
+    const lastCheckedRadio = this.#radioElements.findLast(
+      ({ checked }) => checked,
+    );
+
+    if (lastCheckedRadio) {
+      this.#value = lastCheckedRadio.value;
+
+      if (!this.disabled) {
+        lastCheckedRadio.tabIndex = 0;
       }
-    }
 
-    const checkedRadio = this.value
-      ? this.#radioElements.find(({ value }) => value === this.value)
-      : this.#radioElements.find(
-          ({ checked, disabled }) => checked && !disabled,
-        );
+      // When there's not a default `value` set on the group and a radio is checked
+      // set, we need to manually set the `value` attribute on the group.
+      //
+      // This way when `form.reset()` is called we know what value to reset
+      // back to.
+      //
+      // Guarding against the value being an empty string ensures this component's
+      // `value` setter, called after the attribute is set, doesn't uncheck the radio.
+      if (lastCheckedRadio.value !== '') {
+        this.setAttribute('value', lastCheckedRadio.value);
+      }
 
-    // When there isn't a default `value` attribute set on the group, but a
-    // child radio has `checked` set instead, we need to manually set the
-    // `value` attribute on the group.
-    //
-    // This way when `form.reset()` is called we know what value to reset
-    // back to.
-    if (!this.value && checkedRadio?.value) {
-      this.setAttribute('value', checkedRadio.value);
+      return;
```

Comment:
This early return pattern only works well when it's done consistently through a method. So, in addition to the bugfixes, I've reworked this method so that each condition returns. Probably easier to see if you view the file without the diff.

Reply:
Good call.  I did find [reading `firstUpdated()` here](https://github.com/CrowdStrike/glide-core/blob/63ed6809ef4d35df44c1c8073b6054784f859de8/src/radio-group.ts#L194-L280) easier to follow 👍